File: AndroidStudio/SimpleAdapter/app/src/main/java/to/msn/wings/simpleadapter/MainActivity.java
package to.msn.wings.simpleadapter;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.ListView;
import android.widget.SimpleAdapter;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //リストに表示するデータを準備
        String[] titles = {"革命のエチュード","G線上のアリア","シャコンヌ","夜の女王のアリア","春の海"};
        String[] tags = {"ピアノ","バイオリン","チェロ","声楽","箏"};
        String[] details = {"detail1","detail2","detail3","detail4","detail5"};

            //配列の内容をHashMapに詰め替え
        ArrayList<HashMap<String,String>> data = new ArrayList<>();
        for(int i=0; i<titles.length;i++){
            HashMap<String,String> item = new HashMap<>();
            item.put("title",titles[i]);
            item.put("tag",tags[i]);
            item.put("detail",details[i]);
            data.add(item);
        }
            //HashMap配列とレイアウトを関連付け
        SimpleAdapter simpleAdapter = new SimpleAdapter(this, data,R.layout.list_item,
                new String[]{"title","tag","detail"},
                new int[]{R.id.title,R.id.tag,R.id.detail});

            //アダプターをもとにリストを生成
        ListView listView = findViewById(R.id.list);
        listView.setAdapter(simpleAdapter);
    }
}

File: AndroidStudio/AndroidPractice_Kotlin/MediaSample/app/src/main/java/to/msn/wings/mediasample/MainActivity.kt
package to.msn.wings.mediasample

import android.media.MediaPlayer
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.CompoundButton
import android.widget.Switch
import kotlinx.android.synthetic.main.activity_main.*
import java.io.IOException
import java.lang.IllegalArgumentException

class MainActivity : AppCompatActivity() {

    //メディアプレイヤーフィールド
    private var player: MediaPlayer? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

            //メディアプレーヤーオブジェクトを生成
        player = MediaPlayer()
        //音声ファイルのURI文字列を作成
        val mediaUriStr = "android.resource://${packageName}/${R.raw.mountain_stream}"
        //URIオブジェクトを生成
        val mediFileUri = Uri.parse(mediaUriStr)
        try{
            //メディアプレイヤーに音声ファイルを指定。
            player?.setDataSource(applicationContext,mediFileUri)
            //非同期でのメディア再生準備が完了した際のリスナを設定
            player?.setOnPreparedListener(PlayerPreparedListener())
            //メディア再生が終了したときのリスナを設定
            player?.setOnCompletionListener (PlayerCompletionListener())
            player?.prepareAsync()
        }catch (ex : IllegalArgumentException){
            Log.e("MediaSample", "例外発生",ex)
        }catch (ex: IOException){
            ex.printStackTrace()
        }
        val loopSwicth = findViewById<Switch>(R.id.swLoop)
        loopSwicth.setOnCheckedChangeListener(LoopSwicthChangedListener())

    }

    fun onPlayButtonClick(view:View){
        player?.let {
            //再生ボタンを取得
            val btPlay = findViewById<Button>(R.id.btPlay)
            if(it.isPlaying){
                it.pause()
                btPlay.setText(R.string.bt_play_play)
            }else{
                it.start()
                btPlay.setText(R.string.bt_play_pause)
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        player?.let {
            if(it.isPlaying){
                it.stop()
            }
            it.release()
            player = null
        }
    }

    private inner class PlayerPreparedListener : MediaPlayer.OnPreparedListener{
        override fun onPrepared(mp: MediaPlayer?) {
            val btPlay = findViewById<Button>(R.id.btPlay)
            btPlay.isEnabled = true
            val btBack = findViewById<Button>(R.id.btBack)
            btBack.isEnabled = true
            val btForward = findViewById<Button>(R.id.btForward)
            btForward.isEnabled = true
        }
    }

    private inner class PlayerCompletionListener : MediaPlayer.OnCompletionListener{
        override fun onCompletion(mp: MediaPlayer?) {
            //再生ボタンのラベルを再生に設定
            val btPlay = findViewById<Button>(R.id.btPlay)
            btPlay.setText(R.string.bt_play_play)
        }
    }

    fun onBackButtonClick(view:View){
        player?.seekTo(0)
    }

    fun onForwardButtonClick(view:View){
        player?.let{
            val duration = it.duration
            it.seekTo(duration)
            if(!it.isPlaying){
                it.start()
            }
        }
    }

    private inner class LoopSwicthChangedListener : CompoundButton.OnCheckedChangeListener{
        override fun onCheckedChanged(buttonView: CompoundButton?, isChecked: Boolean) {
            player?.isLooping = isChecked
        }
    }

}
